commits.go

```diff
@@ -261,3 +294,107 @@ func (i *commitsByHashIter) nextList() (*object.Commit, error) {
 		return commit, nil
 	}
 }
+
+type commitIndexKey struct {
```

Comment:
Could we have a generic `ObjectIndexKey` struct for all the objects that are into the packfile?